### ISSUE TITLE
Please add Return and exchange function

### ISSUE THREAD

**Issue by alibler** [Dolibarr/dolibarr#16491]
# Instructions

# Feature Request
As a mature ERP system, The return and exchange function should be built in the completed sales order or purchase order.

## Use case

![Replacmet-Order-or-Return-Order](https://user-images.githubusercontent.com/17175103/109445888-8708d080-7a7b-11eb-9a1e-7fcf99fe41e6.png)

**Comment by dolibarr95** [comment]
Hi :)
Is it the same request ?
https://github.com/Dolibarr/dolibarr/issues/12200

**Comment by alibler** [comment]
It's a similar request, but why hasn't dolibarr officially provided this basic function for so many years? Axel dev development of the return module is very simple! His return module doesn't have the function of tracking the order in the regular ERP system at all.

How do you realize the process management of sales return or purchase return?

**Comment by atm-maxime** [comment]
Hi. Dolibarr is developed by an open-source community and made to be used by most companies and organization. The fact that the "return" function isn't existing yet is just that no one started to work on it. Everyone wants Dolibarr to already have its most favorite function, but developments are made by volunteers or financed by users. And there are not many requests about returns. In the meantime, using an external module such as the one you mention is a possibility. Doing manual stock movements and credit note is also possible.

If the feature request is already fully explained in #12200v, please close this duplicate one.

**Comment by BebZ** [comment]
Yes, it may look "simple" for integrators that are always wondering why Dolibarr doens't make coffee... 
For developpers they will not make a a "simple" module (as existing on dolisotre maybe). 

It has to be considered as a new Object, like propal/order/invoice/etc... 
and not only linked to a "stock movement" and "credit note"... 
else it can never be complete and linked to all convenient objects like  projects, accounting, banks, permissions, etc...
and it will never be accepted in the "core" I think if it is not a very clean and complete object. Am I wrong ?

**Comment by aljawaid** [comment]
I had the Product Returns module made by @AXeL-dev to solve my small business problem as mentioned in the intial post by @alibler so I do agree that Dolibarr needs this in the core.

I do also agree with @atm-maxime on the Dolibarr explanation.

In support of the module, I must say if it helps before purchasing, the reason i made this module was for my small business (physical store).  I wanted a simple product return system where i could log my products with returns on a custom reference number. I could also add notes and statuses to the returned product and have an 'exchanged' status too for auditing purposes.  The lists works fine and the returned product entries can be linked to supplier invoices.  Yes, you need to go to the product returns section to see the module in action (meaning it doesnt link the supplier invoice > return both ways) but I found this easier to manage. Stock can also be moved using stock movement.

### **Create Return**
![image](https://user-images.githubusercontent.com/10233708/139218153-c8b85f15-3e47-47b1-996d-93a6745daad6.png)

### **Completed Return**
![image](https://user-images.githubusercontent.com/10233708/139219062-f761d1f4-ddc2-4294-8d83-0600e06873ae.png)

**Comment by DATUC** [comment]
Hi @ll This module need to e added in the core this is neded since more than 4years allready and to be honest is true the Axel modul is ther but lake many functions and @aljawaid wil note be on stage 7/7 to manage products returns and this is not cool

**Comment by DATUC** [comment]
Hi @ll any news about this ???!!!...

**Comment by dolibarr95** [comment]
Hi!
still relevant

**Comment by DATUC** [comment]
> Hi. Dolibarr is developed by an open-source community and made to be used by most companies and organization. The fact that the "return" function isn't existing yet is just that no one started to work on it. Everyone wants Dolibarr to already have its most favorite function, but developments are made by volunteers or financed by users. And there are not many requests about returns. In the meantime, using an external module such as the one you mention is a possibility. Doing manual stock movements and credit note is also possible.
> 
> If the feature request is already fully explained in #12200v, please close this duplicate one.

Hi all @atm-maxime **And there are not many requests about returns** i'm not ok with this anser this is hardly needed more than 7 years allready that many users are mouving out from Dolibarr becaus eof this not ben availabel other are tring manual management and correction or retur using other way not not the proper way and at time it end up with a big mess

**Comment by YannisWill** [comment]
Regarding the issue of sales returns and procurement returns, this problem has been ongoing for six years since its submission and the core functional requirement remains unresolved. Dear official engineers, when can we expect this issue to be addressed?

**Comment by YannisWell** [comment]
### **Regarding the issue of sales returns and procurement returns, this problem has been ongoing for six years since its submission and the core functional requirement remains unresolved. Dear official engineers, when can we expect this issue to be addressed?**https://github.com/Dolibarr/dolibarr/issues/12200

**Comment by WILLIBBY** [comment]
+1
